test-utils/state-viewer/src/commands.rs

```diff
@@ -50,6 +52,37 @@ pub(crate) fn state(home_dir: &Path, near_config: NearConfig, store: Arc<Store>)
     }
 }
 
+pub(crate) fn dump_contracts(
+    output: &Path,
+    home_dir: &Path,
+    near_config: NearConfig,
+    store: Arc<Store>,
+) {
+    let (runtime, state_roots, header) = load_trie(store, home_dir, &near_config);
+    println!("Storage roots are {:?}, block height is {}", state_roots, header.height());
+    let mut distinct_codes: HashMap<Vec<u8>, AccountId> = HashMap::default();
+    for (shard_id, state_root) in state_roots.iter().enumerate() {
+        let trie = runtime.get_trie_for_shard(shard_id as u64, header.prev_hash()).unwrap();
+        let trie = TrieIterator::new(&trie, state_root).unwrap();
+        // Optimization: contract nodes form a contiguous segment in the set of nodes with values, so if we touched a
+        // contract node and then touched a non-contract node, then we can stop iterating.
+        let mut touched_contract_node = false;
+        for item in trie {
+            let (key, value) = item.unwrap();
+            if is_contract_code_key(&key) {
+                touched_contract_node = true;
+                let account_id = parse_account_id_from_contract_code_key(&key).unwrap();
+                distinct_codes.insert(value.clone(), account_id.clone());
+            } else if touched_contract_node {
+                break;
+            }
+        }
+    }
+    let mut codes_to_dump: HashMap<_, _> =
+        distinct_codes.iter().map(|(code, account_id)| (account_id, to_base64(code))).collect();
+    std::fs::write(output, serde_json::to_string_pretty(&codes_to_dump).unwrap());
```

Comment:
You probably want to `std::fs::create_dir_all(output.parent())` for convenience.